src/Newsletter/Queue/DefaultNewsletterQueue.php

```diff
@@ -92,13 +92,15 @@ public function executeImmidiateAsyncQueueItems()
         }
 
         $this->getLogger()->info('execute immidiate async queue items');
-
-        foreach($this->immidateAsyncQueueItems as $item) {
-            $php = Console::getExecutable('php');
-            $cmd = sprintf($php . ' ' . PIMCORE_PROJECT_ROOT . "/bin/console cmf:newsletter-sync --process-queue-item='%s'", $item->toJson());
-            $this->getLogger()->info('execute async process queue item cmd: ' . $cmd);
-            Console::execInBackground($cmd);
-        }
+        
+        if (count($this->immidateAsyncQueueItems) <= 1) { //avoid too many parallel scripts running
```

Comment:
what is the idea of having a if that checks for count 1 and after that a foreach? 
wouldn't it be the same to call `$item = $this->immidateAsyncQueueItems[0]`? 
In my opinion that would much more readable....

Reply:
Idea of the current statement:
- If there is exactly one customer update (typically the case when a customer is saved in the UI), then this customer will be processed. 
- If there are multiple customer updates that should be processed (for instance on termination of a batch script), then no customer will be processed in parallel, but the queue processor will take care of them all.

Your statement:
- if any customer has been updated, then the first customer will be processed in a parallel script immediatelly, while the others will be processed later in the queue.
-  so if 5.000 customer updates happened, then one update will be processed by a backend CLI script, while the other 4.999 will be processed later in the queue.

Reply:
But yes, maybe we can get rid of the entire foreach loop.